cylc/flow/data_store_mgr.py

```diff
@@ -1380,7 +1380,7 @@ def insert_job(self, name, cycle_point, status, job_conf):
             name=tproxy.name,
             cycle_point=tproxy.cycle_point,
             execution_time_limit=job_conf.get('execution_time_limit'),
-            platform=job_conf.get('platform')['name'],
+            platform=job_conf['platform']['name'],
```

Comment:
Unrelated, spotted during testing.

The `.get` here wasn't protecting against anything as the default return value of `None` would cause the `['name']` bit to error anyway.